index.bs

```diff
@@ -3331,6 +3331,207 @@ be the same.
 </div>
 
 
+<h5 id="idl-overloading-vs-union">Overloading vs. union types</h5>
+
+<i>This section is informative.</i>
+
+For specifications defining IDL [=operations=], it may seem that [=overloaded|overloads=] and a
+combination of [=union types=] and [=optional argument=] have some feature overlap.
+
+It is first important to note that [=overloaded|overloads=] may have different behaviors than
+[=union types=] or [=optional arguments=], and one <em>cannot</em> be fully defined using the
+other (unless, of course, additional prose is provided, which can defeat the purpose of the Web IDL
+type system). For example, given the {{CanvasDrawPath/stroke()}} operations defined on the
+{{CanvasDrawPath}} interface [[HTML]]:
+
+<pre highlight="webidl">
+    interface CanvasDrawPath {
+      void stroke();
+      void stroke(Path2D path);
+    };
+</pre>
+
+Per the ECMAScript language binding, calling <code>stroke(undefined)</code> on a object implementing
+{{CanvasDrawPath}} would throw a {{TypeError}}. However, if the operations were instead defined with
+[=optional arguments=] and merged into one:
+
+<pre highlight="webidl">
+    interface CanvasDrawPath {
+      void stroke(optional Path2D path);
+    };
+</pre>
+
+The [=overload resolution algorithm=] would treat the |path| argument as not present given the same
+ECMAScript language code <code>stroke(undefined)</code>, and not throw any exceptions.
+
+Additionally, there are semantic differences as well. [=Overloaded=] operations are designed to map
+well to C++ overloading at the language binding, and are usually a better fit for operations with
```

Comment:
Not sure this is "at the language binding", but the sentence is true and this is good to point out...